packages/qwik/src/core/shared/component-execution.ts

```diff
@@ -101,7 +102,13 @@ export const executeComponent = (
       (jsx) => {
         const useOnEvents = container.getHostProp<UseOnMap>(renderHost, USE_ON_LOCAL);
         if (useOnEvents) {
-          return maybeThen(addUseOnEvents(jsx, useOnEvents), () => jsx);
+          let _jsx = jsx;
+          if (_jsx && (_jsx as JSXNodeInternal<FunctionComponent>).type === Slot) {
+            _jsx = _jsxSorted(Fragment, null, null, [jsx], 0, null);
+          }
```

Comment:
We should do it in a different cleaner way. We already do such as comparisons in the `addScriptNodeForInvisibleComponents`.
Probably we need to change the `addScriptNodeForInvisibleComponents` or the `addUseOnEvent`.

Reply:
done